src/Lean/Data/Lsp/Extra.lean

```diff
@@ -37,6 +37,7 @@ instance : ToJson WaitForDiagnostics :=
 
 structure LeanFileProgressProcessingInfo where
   range : Range
+  error : Bool := false
```

Comment:
Would it make sense to go for an enum here instead (which would allow us to add other states in the future)?
```lean
inductive LeanFIleProgressKind
  | processing
  | importError
  deriving FromJson, ToJson
```

Reply:
Sounds good to me. A potential further state could be "not processed because there was a `#exit` above" (in grey, perhaps?).

Reply:
Good idea. The only problem I see is that we report the progress as an `Array` of `LeanFileProgressProcessingInfo`. How would we aggregate multiple progresses with different kinds? I'm currently using `any`/`some` to search for an error somewhere in the array in the linked extension PR.

Reply:
> How would we aggregate multiple progresses with different kinds?

Just paint each range separately. The "represent progress by the beginning line number" design is a hack that dates back to the time where the extension mined the diagnostics for progress information.  (The server used to send diagnostics with "processing..." text at the line it was currently at.)

The old Lean 3 extension also painted each range separately and it worked just fine: https://github.com/leanprover/vscode-lean/blob/faa1d9c76406f1ed31e62ee942505328be5908c3/src/taskgutter.ts#L27-L39

Reply:
Ah, I see, I will change it to an enum then.